src/shell/shell-primary-bar.test.tsx

```diff
@@ -446,4 +446,90 @@ describe('Shell primary bar', () => {
 
 		expect(queryByRoleWithIcon('button', { icon: ICONS.settings })).not.toBeInTheDocument();
 	});
+
+	describe('Primary Badge', () => {
+		test('should render the counter badge if show and showCount are true and count is greater than or equal to 0', () => {
```

Comment:
Checking the code again, I think we should just say that the counter badge is shown if count is valued. Negative numbers are accepted too.